railties/lib/rails/application/configuration.rb

```diff
@@ -348,6 +348,11 @@ def load_defaults(target_version)
           Regexp.timeout ||= 1 if Regexp.respond_to?(:timeout=)
         when "8.1"
           load_defaults "8.0"
+
+          # Development and test environment tend to reload code and
+          # redefine methods (e.g. mocking), hence YJIT isn't generally
+          # faster in these environments.
+          self.yjit = !Rails.env.local?
```

Comment:
`!Rails.env.local?` is potentially contentious as it makes it harder to introduce environemnts other than the default ones.

Another possibility is to use `!config.eager_loading`, but this would enable YJIT on CI with the default config.

Reply:
> `!Rails.env.local?` is potentially contentious as it makes it harder to introduce environemnts other than the default ones.

How about an explicit setting on both: `test.rb` and `development.rb` with a proper description? Developers will have better insight into when and how YJIT is not enabled, allowing them to introduce more custom "local" environments.

> Another possibility is to use `!config.eager_loading`, but this would enable YJIT on CI with the default config.

I can't say that I like this idea. It is not unheard of to set `config.eager_load = ENV["EAGER_LOAD"].present?` and run the test suite with `EAGER_LOAD=true bin/test` to ensure there are no syntax errors in files that aren't being tested and ensure everything can be loaded. This would tie seemingly unrelated configurations.